kotlintest-core/src/main/kotlin/io/kotlintest/extensions/TestListener.kt

```diff
@@ -4,26 +4,126 @@ import io.kotlintest.Description
 import io.kotlintest.Spec
 import io.kotlintest.TestCase
 import io.kotlintest.TestResult
+import io.kotlintest.IsolationMode.*
 
 interface TestListener {
 
   /**
-   * This function will be invoked each time a new [TestCase] is executed.
+   * Is invoked once the Test Engine is started.
+   */
+  fun beforeProject(): Unit = Unit
+
+  /**
+   * Is invoked once the Test Engine has finished.
+   */
+  fun afterProject(): Unit = Unit
+
+  /**
+   * This callback will be invoked before a [TestCase] is executed.
    *
-   * @param description the [Description] for the [TestCase] instance.
+   * If a test case is inactive (disabled), then this method will not
```

Comment:
Wouldn't this work similarly to `afterTest`? If yes, then the description should be similar